src/Aspire.Hosting.Azure.CosmosDB/AzureCosmosDBExtensions.cs

```diff
@@ -106,28 +112,45 @@ private static IResourceBuilder<AzureCosmosDBResource> RunAsEmulator(this IResou
                 throw new InvalidOperationException("CosmosClient is not initialized.");
             }
 
-            await cosmosClient.ReadAccountAsync().WaitAsync(ct).ConfigureAwait(false);
-
-            foreach (var database in cosmosDb.Databases)
+            try
             {
-                var db = (await cosmosClient.CreateDatabaseIfNotExistsAsync(database.DatabaseName, cancellationToken: ct).ConfigureAwait(false)).Database;
+                await cosmosClient.ReadAccountAsync().WaitAsync(ct).ConfigureAwait(false);
 
-                foreach (var container in database.Containers)
+                foreach (var database in cosmosDb.Databases)
                 {
-                    var containerProperties = container.ContainerProperties;
+                    var db = (await cosmosClient.CreateDatabaseIfNotExistsAsync(database.DatabaseName, cancellationToken: ct).ConfigureAwait(false)).Database;
+
+                    foreach (var container in database.Containers)
+                    {
+                        var containerProperties = container.ContainerProperties ?? new ContainerProperties
+                        {
+                            Id = container.ContainerName,
+                            PartitionKeyPaths = container.PartitionKeyPaths
+                        };
 
-                    await db.CreateContainerIfNotExistsAsync(containerProperties, cancellationToken: ct).ConfigureAwait(false);
+                        await db.CreateContainerIfNotExistsAsync(containerProperties, cancellationToken: ct).ConfigureAwait(false);
+                    }
                 }
+                creationState = HealthCheckResult.Healthy();
```

Comment:
This isn't really how health checks are supposed to work. They should be checking that the resource can be connected to correctly. This is saying once the db/containers are all created, assume it is healthy. But there may be other connection issues.